src/network.toit

```diff
@@ -49,13 +54,23 @@ class EndpointHttp implements Endpoint:
       // firmware if requested to do so.
       validate-firmware --reason="connected to network"
 
+      request-mutex := monitor.Mutex
```

Comment:
It feels a bit weird to use a mutex here. You just want to delay the shutting down and you end up having `socket.close` in two positions.